jupyterlab_bigquery/src/components/list_items_panel/list_tree_item.tsx

```diff
@@ -426,14 +409,18 @@ export class DatasetResource extends Resource<DatasetProps> {
     };
     try {
       this.setState({ loading: true });
-      await listTablesService.listTables(dataset.id).then((data: Dataset) => {
-        newDataset.tables = data.tables;
-        newDataset.tableIds = data.tableIds;
-      });
-      await listModelsService.listModels(dataset.id).then((data: Dataset) => {
-        newDataset.models = data.models;
-        newDataset.modelIds = data.modelIds;
-      });
+      await bigQueryService
```

Comment:
Since you're using `await` here, the listTables and listModels API calls will run sequentially when it looks like you can probably run them concurrently and improve the overall performance. 

There's sort of an odd blend of newer async/await style Promise handling along with older `.then(` based handling. The latter is actually probably bit more intuitive in this case where you want each Promise to run its asynchronous code concurrently and then update the state when it finishes.

At a high level, I think you want to issue both requests, handle errors in either one without affecting the other, and when both have completed (either successfully or with error) set the loading state to false.